fastly/block_fastly_service_v1_gzip.go

```diff
@@ -173,6 +175,33 @@ func (h *GzipServiceAttributeHandler) Read(d *schema.ResourceData, s *gofastly.S
 
 	gl := flattenGzips(gzipsList)
 
+	// Note: Although "content_types" and "extensions" fields are optional in spec,
+	// Fastly API will actually set the default value silently when these fields are not sent
+	// or an empty field value is sent. This will cause unexpected diff.
+	// We need to ignore these fields in the API response unless field values are explicitly set.
+	type IgnoreFields struct {
+		Name string
+	}
+	ignoreList := map[string][]IgnoreFields{}
+
+	for _, elem := range d.Get("gzip").(*schema.Set).List() {
+		name := elem.(map[string]interface{})["name"].(string)
```

Comment:
Might be nice to do the type assertion only once rather than three times:

```go
m := elem.(map[string]interface{})
name := m["name"].(string)
if m["content_types"].(*schema.Set).Len() == 0 {
	ignoreList[name] = append(ignoreList[name], IgnoreFields{Name: "content_types"})
}
if m["extensions"].(*schema.Set).Len() == 0 {
	ignoreList[name] = append(ignoreList[name], IgnoreFields{Name: "extensions"})
}
```